dcos_test_utils/upgrade.py

```diff
@@ -125,17 +127,30 @@ def upgrade_dcos(
                     '--speed-limit', '100000',
                     '--speed-time', '60',
                     '--remote-name', upgrade_script_path])
-            ssh_client.command(host, ['sudo', 'bash', 'dcos_node_upgrade.sh'])
-            wait_metric = {
-                'master': 'registrar/log/recovered',
-                'slave': 'slave/registered',
-                'slave_public': 'slave/registered',
-            }[role]
-            log.info('Waiting for {} to rejoin the cluster...'.format(role_name))
-            try:
-                wait_for_mesos_metric(dcos_api_session, host, wait_metric)
-            except retrying.RetryError as exc:
-                raise Exception(
-                    'Timed out waiting for {} to rejoin the cluster after upgrade: {}'.
-                    format(role_name, repr(host))
-                ) from exc
+            if use_checks:
+                try:
+                    output = ssh_client.command(host, ['sudo', 'bash', 'dcos_node_upgrade.sh'], stdout="")
+                    log.info("Output from checks" + output)
+                except CalledProcessError as e:
+                    returncode = e.returncode
+                    if returncode != 0:
+                        raise Exception(
+                            'Timed out waiting for {} to rejoin the cluster after upgrade: {}'.
```

Comment:
I don't see what makes this a timeout error specifically. How is nonzero guaranteed to be a timeout? Is there no other failure case?